src/actions/AuthActions.js

```diff
@@ -25,38 +25,42 @@ const attemptFacebookLogin = async (dispatch) => {
 
     if (type === 'success') {
         let {data: {email, name, picture: {data: {url}}}} = await axios.get(`https://graph.facebook.com/me?fields=id,name,email,picture&access_token=${token}`);
-        // we now have access to email, name, and their profile picture
-
-        await AsyncStorage.setItem(tokens.USER_TOKEN, token);
 
         dispatch({
             type: FACEBOOK_LOGIN_SUCCESS,
-            payload: token
-        })
+            payload: {token, appReady: false}
+        });
+
+        await AsyncStorage.setItem(tokens.USER_TOKEN, token);
+
+        navigate(navKeys.SEARCH);
     } else {
         dispatch({
             type: FACEBOOK_LOGIN_FAILURE
-        })
+        });
+
+        navigate(navKeys.LOGIN);
     }
 };
 
-export const logUserIntoFacebook = () => {
+export const logUserIntoFacebook = ({navigate}) => {
 
     return async (dispatch) => {
+        navigate(navKeys.AUTH);
```

Comment:
When a user begins the FB login process, the first thing we do is send them to the `AuthScreen`. Then, the result of the FB login will dictate where they're sent to after.